### ISSUE TITLE
rustc 1.13.0 hangs/out of memory

### ISSUE THREAD

**Issue by ctsa** [rust-lang/rust#37942]
I'm using the latest stable release (1.13.0) installed via rustup on Ubuntu 16.04. I ran into a compiler hang after a copy/paste slipup, and reduced the testcase down to the following:

main.rs:
```rust
fn main() {
    extern crate rand;
    use rand::Rng;
}
```

What I see when compiling is:
```
$ /usr/bin/time -v rustc main.rs 
fatal runtime error: out of memory
Command exited with non-zero status 1
        Command being timed: "rustc main.rs"
        User time (seconds): 47.28
        System time (seconds): 0.98
        Percent of CPU this job got: 99%
        Elapsed (wall clock) time (h:mm:ss or m:ss): 0:48.56
        Average shared text size (kbytes): 0
        Average unshared data size (kbytes): 0
        Average stack size (kbytes): 0
        Average total size (kbytes): 0
        Maximum resident set size (kbytes): 4259424
        Average resident set size (kbytes): 0
        Major (requiring I/O) page faults: 0
        Minor (reclaiming a frame) page faults: 88942
        Voluntary context switches: 9
        Involuntary context switches: 3471
        Swaps: 0
        File system inputs: 0
        File system outputs: 0
        Socket messages sent: 0
        Socket messages received: 0
        Signals delivered: 0
        Page size (bytes): 4096
        Exit status: 1
```

**Comment by durka** [comment]
Fixed on beta/nightly.

**Comment by arielb1** [comment]
cc @jseyfried

**Comment by jseyfried** [comment]
This is #36881, which was fixed in #36882 (currently in beta).
